app/lib/till.rb

```diff
@@ -0,0 +1,47 @@
+require_relative 'order'
+require_relative 'receipt'
+require_relative 'payment'
+require_relative 'menu'
+
+
+class Till
+
+  include Menu
+
+  attr_reader :order, :receipt, :tax, :change
+
+  def initialize
+    @tax = 8.64
+    menu_read_price_list
+    new_order
+  end
+
+  def new_order
+    @order = Order.new
+    @receipt = nil
+    @change = nil
+  end
+
+  def order_list
+    order.order_list
+  end
+
+  def add_item_to_order(item, quantity)
+    order.add_item(item, quantity)
+  end
+
+  def print_receipt
+    @receipt = Receipt.new(order_list, tax)
+    receipt.print_receipt
+  end
+
+  def total_due
+    @receipt ? receipt.total_due : 'Please generate a receipt'
```

Comment:
Does the receipt carry the total_due? If you had the order calculate this, you would not need to store the receipt as an instance variable, and you wouldn't need this conditional - it would simplify your code!